src/sql/magic.py

```diff
@@ -43,6 +43,23 @@
 
 
 SUPPORT_INTERACTIVE_WIDGETS = ["Checkbox", "Text", "IntSlider", ""]
+recognized_arguments = [
+    "--connections",
+    "--close",
+    "--creator",
+    "--section",
+    "--persist",
+    "--persist-replace",
+    "--no-index",
+    "--append",
+    "--connection_arguments",
+    "--file",
+    "--save",
+    "--with",
+    "--no-execute",
+    "--alias",
+    "--interact",
```

Comment:
I'm sure there's a way to get the declared arguments so we don't have to manually keep a list of them

Reply:
```python
In [1]: from sql.magic import SqlMagic

In [2]: SqlMagic.execute.parser
Out[2]: MagicArgumentParser(prog='execute', usage=None, description='\n        Runs SQL statement against a database, specified by\n        SQLAlchemy connect string.\n\n        If no database connection has been established, first word\n        should be a SQLAlchemy connection string, or the user@db name\n        of an established connection.\n\n        Examples::\n\n          %%sql postgresql://me:mypw@localhost/mydb\n          SELECT * FROM mytable\n\n          %%sql me@mydb\n          DELETE FROM mytable\n\n          %%sql\n          DROP TABLE mytable\n\n        SQLAlchemy connect string syntax examples:\n\n          postgresql://me:mypw@localhost/mydb\n          sqlite://\n          mysql+pymysql://me:mypw@localhost/mydb\n\n        ', formatter_class=<class 'IPython.core.magic_arguments.MagicHelpFormatter'>, conflict_handler='error', add_help=False)

In [3]: from sql.parse import _option_strings_from_parser

In [4]: _option_strings_from_parser(SqlMagic.execute.parser)
Out[4]:
['-l',
 '--connections',
 '-x',
 '--close',
 '-c',
 '--creator',
 '-s',
 '--section',
 '-p',
 '--persist',
 '-P',
 '--persist-replace',
 '-n',
 '--no-index',
 '--append',
 '-a',
 '--connection_arguments',
 '-f',
 '--file',
 '-S',
 '--save',
 '-w',
 '--with',
 '-N',
 '--no-execute',
 '-A',
 '--alias',
 '--interact']

In [5]:
```